### PR TITLE
Update Blazor Server App template name to `BlazorApp`

### PR DESCRIPTION
**Pull request by mkArtakMSFT**

### CODE REVIEW

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/template.json
Comment (javiercn) [inline src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/template.json:448]: We don't abbreviate things on user displayed strings
```suggestion
  "defaultName": "BlazorApplication",
```

Comment (mkArtakMSFT) [reply src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/template.json:448]: @danroth27 I'll let you suggest the exact phrase here.

Comment (danroth27) [reply src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/template.json:448]: App is not an abbreviation. It is a modern synonym for application. Examples of this usage include "Apple App Store". Use is "app" is preferred over "application" in the Microsoft Style Guide: https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/a/app-application.

Comment (HaoK) [reply src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/template.json:448]: Should we update WebApplication to WebApp to then? https://github.com/dotnet/aspnetcore/blob/241b10a43fecc7a08ad875bb4f483db8cc754511/src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/.template.config/template.json#L374

Comment (danroth27) [reply src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/template.json:448]: This change was motivated by the principle of "don't move people's cheese unnecessarily". Blazor apps in VS historically default to the BlazorApp name, but that changed with the move to the new New Project Dialog experience. This PR fixes that and avoids us having to update docs, screen shots, etc. Web apps have historically defaulted to WebApplication1, and I don't see any reason to change that. I brought up the Microsoft Style Guide purely to show that "app" is valid, not that it is always preferred.

**Review by HaoK — APPROVED** [review]

**Comment by mkArtakMSFT** [comment]
@danroth27 I'll wait for you to sign off on this before merging.

**Review by danroth27 — APPROVED** [review]
:shipit:

**Pull request merged**
